sympy/solvers/ode.py

```diff
@@ -5720,6 +5724,110 @@ def checkinfsol(eq, infinitesimals, func=None, order=None):
                     soltup.append((True, 0))
             return soltup
 
+def _ode_lie_group_try_heuristic(eq, heuristic, func, match, inf):
+
+    xi = Function("xi")
+    eta = Function("eta")
+    f = func.func
+    x = func.args[0]
+    y = match['y']
+    h = match['h']
+    tempsol = []
+    if not inf:
+        inf = infinitesimals(eq, hint=heuristic, func=func, order=1, match=match)
```

Comment:
`ValueError` is coming from here.

Reply:
When we will try to find` infinitesimals` and maybe for the heuristic, `infinitesimals` doesn't exists so it will raise `ValueError.`